setup.py

```diff
@@ -30,7 +30,13 @@
 with open("requirements.txt") as fid:
     install_requires = [line.strip() for line in fid if line.strip()]
 
-extras_require = {"recommender": ["numpy"]}
+extras_require = {
+    "recommender": ["numpy"],
+    "show_api_usage": ["graphviz"],
+    "show_memory": ["memory_profiler"],
+    "thumbnail": ["pillow"],
```

Comment:
We should make `pillow` a core requirement because I think we always scale thumbnails

Reply:
Yes I did notice that. We currently only have sphinx as a dependency. Will add pillow and amend docs.